File: services/dispute/src/main/java/com/example/iter/dispute/util/ReportMapper.java
package com.example.iter.dispute.util;

import com.example.iter.auth.api.UserSummary;
import com.example.iter.dispute.domain.entity.Report;
import com.example.iter.dispute.dto.response.ReportDetailResponse;
import com.example.iter.dispute.dto.response.ReportSummaryResponse;
import org.springframework.stereotype.Component;

@Component
public class ReportMapper {

    // 신고와 신고자 정보를 신고 목록 응답으로 변환합니다.
    public ReportSummaryResponse toSummary(Report report, UserSummary reporter) {
        return new ReportSummaryResponse(
                report.getId(),
                reporter,
                report.getTargetType(),
                report.getTargetId(),
                report.getReason(),
                report.getStatus(),
                report.getCreatedAt()
        );
    }

    // 신고와 신고자 정보를 신고 상세 응답으로 변환합니다.
    public ReportDetailResponse toDetail(Report report, UserSummary reporter) {
        return new ReportDetailResponse(
                report.getId(),
                reporter,
                report.getTargetType(),
                report.getTargetId(),
                report.getReason(),
                report.getDescription(),
                report.getStatus(),
                report.getCreatedAt(),
                report.getResolvedAt()
        );
    }
}

File: services/dispute/src/main/kotlin/com/example/iter/dispute/util/ReportMapper.kt
package com.example.iter.dispute.util

import com.example.iter.auth.api.UserSummary
import com.example.iter.dispute.domain.entity.Report
import com.example.iter.dispute.dto.response.ReportDetailResponse
import com.example.iter.dispute.dto.response.ReportSummaryResponse
import org.springframework.stereotype.Component

@Component
class ReportMapper {

    fun toSummary(report: Report, reporter: UserSummary): ReportSummaryResponse = ReportSummaryResponse(
        report.id,
        reporter,
        report.targetType,
        report.targetId,
        report.reason,
        report.status,
        report.createdAt
    )

    fun toDetail(report: Report, reporter: UserSummary): ReportDetailResponse = ReportDetailResponse(
        report.id,
        reporter,
        report.targetType,
        report.targetId,
        report.reason,
        report.description,
        report.status,
        report.createdAt,
        report.resolvedAt
    )
}
